src/plugins/kibana/public/management/index.js

```diff
@@ -42,6 +42,16 @@ uiModules
       if ($scope.section) {
         $scope.section.items.forEach(item => {
           item.class = `#${$location.path()}`.indexOf(item.url) > -1 ? 'active' : undefined;
+          // Disable link if path is not set
+          if (!item.path) {
+            delete item.url;
+          }
+          item.items.forEach(childItem => {
+            // Disable link if path is not set
+            if (!childItem.path) {
+              delete childItem.url;
```

Comment:
Discussed this with @tylersmalley out-of-band so bringing that conversation back into the issue:

We've decided to go a different route altogether, one that we think will simplify the interface to the `ManagementItem` class. We're going to remove the notion of a `path` completely from this interface. Users of this class will need to explicitly specify the `url` instead.